app/Entities/Models/Page.php

```diff
@@ -126,4 +126,18 @@ public function forJsonDisplay(): Page
         $refreshed->html = (new PageContent($refreshed))->render();
         return $refreshed;
     }
+
+    public function getCoverImage(): string
+    {
+        $dom = new \DomDocument();
```

Comment:
I would put the logic to fetch the first page image in `PageContent` class, if that returns null, we can return the book cover.

Reply:
@Abijeet See changes in 48587d2 and let me know your thoughts.